uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java

```diff
@@ -284,10 +292,26 @@ public void execute() {
                                     (PopupActivity) activity);
                 doWhenFinished.execute();
             } else if (activity.isType(ActivityResourceType.PERSPECTIVE.name())) {
-                placeHistoryHandler.flush();
-                launchPerspectiveActivity(place,
-                                          (PerspectiveActivity) activity,
-                                          doWhenFinished);
+                final Command launchPerspectiveCommand = () -> {
+                    placeHistoryHandler.flush();
+                    launchPerspectiveActivity(place,
+                                              (PerspectiveActivity) activity,
+                                              doWhenFinished);
+                };
+
+                final PerspectiveActivity currentPerspective = perspectiveManager.getCurrentPerspective();
+
+                if (currentPerspective != null) {
+                    final BiParameterizedCommand<Command, PlaceRequest> closeChain = this.perspectiveCloseChain.get(currentPerspective.getIdentifier());
```

Comment:
@paulovmr could you please add docs on the chain interface?

Reply:
Done.